source/volume_of_fluid/assembler.cc

```diff
@@ -0,0 +1,667 @@
+
+      data.local_matrix = 0;
+      data.local_rhs = 0;
+
+      //loop over all possible subfaces of the cell, and reset corresponding rhs
+      for (unsigned int f = 0; f < GeometryInfo<dim>::max_children_per_face * GeometryInfo<dim>::faces_per_cell; ++f)
+        {
+          data.local_face_rhs[f] = 0.0;
+          data.local_face_matrices_ext_ext[f] = 0.0;
+          data.face_contributions_mask[f] = false;
+        }
+
+      if (update_from_old)
+        {
+          scratch.finite_element_values[solution_field].get_function_values (this->get_old_solution(),
+                                                                             scratch.old_field_values);
+
+          scratch.finite_element_values[volume_of_fluidN_n].get_function_values (this->get_old_solution(),
+                                                                                 scratch.cell_i_n_values);
+
+          scratch.finite_element_values[volume_of_fluidN_d].get_function_values (this->get_old_solution(),
+                                                                                 scratch.cell_i_d_values);
+        }
+      else
+        {
+          scratch.finite_element_values[solution_field].get_function_values (this->get_solution(),
+                                                                             scratch.old_field_values);
+
+          scratch.finite_element_values[volume_of_fluidN_n].get_function_values (this->get_solution(),
+                                                                                 scratch.cell_i_n_values);
+
+          scratch.finite_element_values[volume_of_fluidN_d].get_function_values (this->get_solution(),
+                                                                                 scratch.cell_i_d_values);
+        }
+
+      // Obtain approximation to local interface
+      for (unsigned int q = 0; q< n_q_points; ++q)
+        {
+          // Init FE field vals
+          for (unsigned int k=0; k<volume_of_fluid_dofs_per_cell; ++k)
+            scratch.phi_field[k] = scratch.finite_element_values[solution_field].value(main_fe.component_to_system_index(solution_component, k), q);
+
+          // Init required local time
+          for (unsigned int i = 0; i<volume_of_fluid_dofs_per_cell; ++i)
+            {
+              data.local_rhs[i] += scratch.old_field_values[q] *
+                                   scratch.finite_element_values.JxW(q);
+              for (unsigned int j=0; j<volume_of_fluid_dofs_per_cell; ++j)
+                data.local_matrix (i, j) += scratch.phi_field[i] *
+                                            scratch.phi_field[j] *
+                                            scratch.finite_element_values.JxW(q);
+            }
+        }
+
+      for (unsigned int face_no = 0; face_no < GeometryInfo<dim>::faces_per_cell; ++face_no)
+        {
+          const unsigned int f_dim = face_no/2; // Obtain dimension
```

Comment:
Please move this comment to the line before, and explain this in more detail. Why is the dimension the face number divided by 2? Is this how deal.II orders faces?